addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java

```diff
@@ -317,52 +334,19 @@ private synchronized void connect() {
         throw new IllegalArgumentException("");
     }
 
-    public synchronized void sendMessage(RFXComMessage msg) throws RFXComException {
-
-        ((RFXComBaseMessage) msg).seqNbr = getNextSeqNumber();
-        byte[] data = msg.decodeMessage();
-
-        logger.debug("Transmitting message '{}'", msg);
-        logger.trace("Transmitting data: {}", DatatypeConverter.printHexBinary(data));
-
-        setResponseMessage(null);
 
+    public void sendMessage(RFXComMessage msg) throws RFXComException {
         try {
-            connector.sendMessage(data);
+            RFXComBaseMessage baseMsg = (RFXComBaseMessage) msg;
+            baseMsg.rawMessage = baseMsg.decodeMessage();
```

Comment:
We should not set the `rawMessage` from outside (that should already have been private imho)

To me it looks like not such a bad trade off if we would have to generate that byte array multiple times in case of retries, if it is a problem we should resolve it in the `RFXComBaseMessage` we should not be touching this inner field.